lib/sender.js

```diff
@@ -29,18 +29,27 @@ var parseAndRespond = function(resBody, callback) {
 function extractRecipient(recipient) {
     var validKeys = ['registrationIds', 'registrationTokens', 'topic', 'notificationKey'];
     var theRecipient;
-
-    validKeys.forEach(function(key) {
-        if (recipient[key]) {
+    var err = null;
+    var key = null;
+    for(var i = 0; i < validKeys.length; i++){
+        key = validKeys[i];
+        if(recipient[key]) {
+            if(!Array.isArray(recipient[key])){
+                err = "Value in key : " + key + " must contain Array";
+                break;
+            }
             theRecipient = recipient[key];
-            return;
+            break;
         }
-    });
+    }
+
+    if(i == validKeys.length){
```

Comment:
in case of invalid key err will be null but i will be equal to the length of validKeys that's why I checked it like this.